x/mongo/driver/topology/topology.go

```diff
@@ -119,6 +119,32 @@ func newServerSelectionState(selector description.ServerSelector, timeoutChan <-
 	}
 }
 
+func New_(cfg *config) (*Topology, error) {
```

Comment:
This will fully replace `topology.New` once we've determine how to handle some edge case test conversions.